exporter/collector/metricsexporter.go

```diff
@@ -97,23 +123,82 @@ func (me *metricsExporter) pushMetrics(ctx context.Context, m pdata.Metrics) err
 			mes := ilm.Metrics()
 			for k := 0; k < mes.Len(); k++ {
 				metric := mes.At(k)
+				for _, md := range me.mapper.metricDescriptor(metric) {
+					me.mds <- md
+				}
 				timeSeries = append(timeSeries, me.mapper.metricToTimeSeries(monitoredResource, extraLabels, metric)...)
 			}
 		}
 	}
 
 	// TODO: self observability
+	// TODO: Figure out how to configure service time series calls.
+	if false {
+		err := me.createServiceTimeSeries(ctx, timeSeries)
+		recordPointCount(ctx, len(timeSeries), m.DataPointCount()-len(timeSeries), err)
+		return err
+	}
 	err := me.createTimeSeries(ctx, timeSeries)
 	recordPointCount(ctx, len(timeSeries), m.DataPointCount()-len(timeSeries), err)
 	return err
 }
 
+// Reads metric descriptors from the md channel, and reports them (once) to GCM.
+func (me *metricsExporter) exportMetricDescriptorRunner() {
+	mdCache := make(map[string]*metricpb.MetricDescriptor)
+	// We iterate over all metric descritpors until the channel is closed.
+	// Note: if we get terminated, this will still attempt to export all descriptors
+	// prior to shutdown.
+	for md := range me.mds {
+		// Not yet sent, now we sent it.
+		// TODO - check to see if this is a service/system metric and doesn't send descriptors.
```

Comment:
FYI - OpenCensus had logic to NOT send metric descriptors on certain domains when we suspect they are "system" (or service) metrics.   I can add that in this CL or a follow on.  It's related to the design of how we want to handle CreateServiceTimeSeries vs. CreateTimeSeries calls.

cc @dashpole

Reply:
That makes sense.  It would be nice if CreateServiceTimeSeries == don't create MD so we can simplify the implementation

Reply:
I'd rather not implement that logic if we don't have to